addon/adapters/osf-adapter.js

```diff
@@ -10,14 +11,82 @@ export default DS.JSONAPIAdapter.extend(DataAdapterMixin, {
     authorizer: 'authorizer:osf-token',
     host: config.OSF.apiUrl,
     namespace: config.OSF.apiNamespace,
-
-    buildURL(modelName, id, snapshot, requestType, query) {  // jshint ignore:line
-        var url = this._super(...arguments);
+    buildURL() {
         // Fix issue where CORS request failed on 301s: Ember does not seem to append trailing
-        //  slash to URLs for single documents, but DRF redirects to force a trailing slash
-        if (url.lastIndexOf('/') !== 0) {
+        // slash to URLs for single documents, but DRF redirects to force a trailing slash
+        var url = this._super(...arguments);
+        if (url.lastIndexOf('/') !== url.length - 1) {
             url += '/';
         }
         return url;
+    },
+    /**
+     * Build the request payload for a relationship create/update. We're
+     * using the meta hash of the relationship field to pass an optional
+     * custom serialization method. This bypasses the normal serialization
+     * flow, but is necessary to cooperate with the OSF APIv2.
+     *
+     * @method _buildRelationshipPayload
+     * @param {DS.Store} store
+     * @param {DS.Snapshot} snapshot
+     * @param {String} relationship the relationship to build a payload for
+     * @return {Object} the serialized relationship
+     **/
+    _relationshipPayload(store, snapshot, relationship) {
+        var relationMeta = snapshot.record[relationship].meta();
+        var relationType = relationMeta.type;
+        var serialized;
+        if (relationMeta.options.serializer) {
+            serialized = relationMeta.options.serializer(snapshot.record);
+        } else {
+            var serializer = store.serializerFor(relationType);
+            var toBeSent = snapshot.record.get(relationship).filter(record => record.id === null);
+            serialized = serializer.serialize(toBeSent);
+        }
+        return serialized;
+    },
+    /**
+     * Construct a URL for a relationship create/update/delete. Has the same
+     * signature as buildURL, with the addition of a 'relationship' param
+     *
+     * @method _buildRelationshipURL
+     * @param {String} relationship the relationship to build a url for
+     * @return {String} a URL
+     **/
+    _buildRelationshipURL() {
```

Comment:
Specifically because this is the definition of the function, we should try to specify the parameters in the call signature- especially since we control the method, and all usages. Can we please avoid using destructuring this way?